nomad/structs/structs.go

```diff
@@ -2152,6 +2160,10 @@ const (
 	// TaskArtifactDownloadFailed indicates that downloading the artifacts
 	// failed.
 	TaskArtifactDownloadFailed = "Failed Artifact Download"
+
+	// TaskDiskExceeded indicates that one of the tasks in a taskgroup has
+	// exceeded the requested disk resources.
+	TaskDiskExceeded = "Disk Resources Exceeded"
```

Comment:
Add a reason field to the `TaskEvent` that can store the disk usage and max allowed as two int64s. Then in `nomad alloc-status` we can pretty print it